### PR TITLE
fix issue 345

### LINKED ISSUE #345: CMakeCache searched wrongly
`CMakeCache.txt` is only searched in `bin` folder.

`fopen Error: file /home/markus/Projekte/Elektra/build/bin/CMakeCache.txt not found`

### PR DESCRIPTION
**Pull request by KurtMi**
fixed #345 and manual invocation improved

### CODE REVIEW

**Comment by KurtMi** [comment]
pdf output failed, due to "Unicode char \u8:ā not set up for use with LaTeX"

**Comment by markus2330** [comment]
Can you try to fix the problem with a LATEX_HEADER (or EXTRA_PACKAGES) that includes `\usepackage[utf8]{inputenc}` or similar.

File: doc/markdownlinkconverter/README.md [hunk 1]
```diff
@@ -9,6 +9,12 @@ The Markdown link Converter gives each markdown file a header `{ #header }` whic
 and converts the links to refer to this headers. This conversion
 happens in 2 passes, which is needed because there can be files with no title.
 
+## Usage for manual invocation
+
+	markdownlinkconverter [<cmake-cache-file>] <input-file>
+
+**The <input-file> parameter must be an absolute path!**
```
Comment (markus2330) [inline doc/markdownlinkconverter/README.md]: Is this relevant anymore if we work with the current working directory?

File: doc/markdownlinkconverter/markdownlinkconverter.c [hunk 2]
```diff
@@ -296,25 +288,35 @@ int main (int argc, char *argv[])
...
-	char executablename[strlen (argv[0]) + 1];
-	char filename[strlen (argv[1]) + 1];
+	char * inputFilename = NULL;
+	char * cmakeCacheFilename = NULL;
 
-	strcpy (executablename, argv[0]);
-	strcpy (filename, argv[1]);
+	if (argc == 2)
+	{
+		inputFilename = alloca (strlen (argv[1]) + 1);
```
Comment (markus2330) [inline doc/markdownlinkconverter/markdownlinkconverter.c]: alloca is not portable, please prefer `char inputFilename[strlen(...)...]`

File: doc/markdownlinkconverter/markdownlinkconverter.c [hunk 3]
```diff
@@ -444,31 +446,25 @@ static void printConvertedPath (FILE * output, char * path)
...
-	while (folderDelimiterToBuildRoot > countDelimToBuildRoot)
+	FILE * input = NULL;
+	if (cmakeCacheFilename)
 	{
-		if (*endOfExecutablename == FOLDER_DELIMITER)
+		input = fopen (cmakeCacheFilename, "r");
+		if (!input)
 		{
-			++countDelimToBuildRoot;
+			fprintf (stderr, "Warning: CmakeCacheFile %s not found\n", cmakeCacheFilename);
```
Comment (markus2330) [inline doc/markdownlinkconverter/markdownlinkconverter.c]: Warning? Isn't this an error?

**Comment by markus2330** [comment]
Thanks for the fix! Please ping me if you are ready.

File: doc/markdownlinkconverter/README.md (hunk 1 above)
Comment (KurtMi) [inline doc/markdownlinkconverter/README.md]: It's nice to have the possibility to specify the "<cmake-cache-file>", doxygen will execute the markdown link converter with 2 parameter anyway.

**Comment by KurtMi** [comment]
> Can you try to fix the problem with a LATEX_HEADER (or EXTRA_PACKAGES) that includes \usepackage[utf8]{inputenc} or similar?

I see two possibilities, the first is `EXTRA_PACKAGES=utf8x`, the second write one package containing `\DeclareUnicodeCharacter` command for each special character we have (currently 1).
If we write an elektratex package it must be copied in the doc/latex folder, as far I know.

**Comment by markus2330** [comment]
Thanks for taking care of the utf8 issue! Please ping me when you have updated this PR.

File: doc/markdownlinkconverter/elektraSpecialCharacters.sty [hunk 4]
```diff
@@ -0,0 +1 @@
+\DeclareUnicodeCharacter{0101}{\=a}
```
Comment (KurtMi) [inline doc/markdownlinkconverter/elektraSpecialCharacters.sty]: Is this the right place for the latex "package" ?

**Comment by KurtMi** [comment]
Build fails at other non utf8 character, but I tested it local, with #342 it works fine.

**Comment by KurtMi** [comment]
Both #350 and #342 should be fine

**Comment by markus2330** [comment]
Thank you!

Seems that different latex have different utf8 characters already declared.

Maybe we should add some other annoying-to-find characters, the various whitespaces and dashes?

Now we have 610 pages ;)

Most Chapters are still named README. Maybe we can rename them to the folder the README.md is in?

**Comment by KurtMi** [comment]
> Maybe we should add some other annoying-to-find characters, the various whitespaces and dashes?

I think people should watch there encoding and only if it's not possible add a `DeclareUnicodeCharacter` in the `elektraSpecialCharacters.sty`. 

> Now we have 610 pages ;)
> Most Chapters are still named README. Maybe we can rename them to the folder the README.md is in?

I will have a look at this after the release.

**Pull request merged**